vhost_user_fs/src/server.rs

```diff
@@ -126,6 +134,95 @@ impl<F: FileSystem + Sync> Server<F> {
         }
     }
 
+    fn setupmapping(
+        &self,
+        in_header: InHeader,
+        mut r: Reader,
+        w: Writer,
+        vu_req: Option<&mut SlaveFsCacheReq>,
+    ) -> Result<usize> {
+        if vu_req.is_none() {
+            return reply_error(
+                io::Error::from_raw_os_error(libc::EINVAL),
+                in_header.unique,
+                w,
+            );
+        }
+
+        // get offset from reader
+        let SetupmappingIn {
+            fh,
+            foffset,
+            len,
+            flags,
+            moffset,
+        } = r.read_obj().map_err(Error::DecodeMessage)?;
+
+        match self.fs.setupmapping(
+            Context::from(in_header),
+            in_header.nodeid.into(),
+            fh.into(),
+            foffset,
+            len,
+            flags,
+            moffset,
+            vu_req.unwrap(),
```

Comment:
Don't `unwrap()` an option like this, please try if it is `Some()`, using the following kind of syntax:
```
if let Some(vu) = vu_req {
...
}
```